app/jinja_filters.py

```diff
@@ -75,11 +86,12 @@ def format_unit(
     value: int | float | Decimal,
     length: UnitLengthType = "short",
 ) -> str:
-    formatted_unit: str = units.format_unit(
+    locale = flask_babel.get_locale()
+    formatted_unit: str = custom_format_unit(
         value=value,
         measurement_unit=unit,
         length=length,
-        locale=flask_babel.get_locale(),
+        locale=locale,
```

Comment:
I thought it was good as in the currency one with this just passed in instead of making a variable